torchrl/data/utils.py

```diff
@@ -35,6 +37,86 @@
 INDEX_TYPING = Union[None, int, slice, str, Tensor, List[Any], Tuple[Any, ...]]
 
 
+def _consolidate_entries_td(
```

Comment:
i kept it for working with TensorDictBase for now because I think that coding this function now to work with lists of TensorDictBases would be have uneeded complexity to handle the various nesting possibbilities (interleaving of lists and LazyStacks).

I think it will be easier to code this with lists of tensordicts once we know there cannot be lazy stack entries.